lib/java/persistance/src/main/java/org/enso/persist/PerGenerator.java

```diff
@@ -31,7 +30,7 @@ private PerGenerator(
   }
 
   static byte[] writeObject(Object obj, Function<Object, Object> writeReplace) throws IOException {
-    var histogram = PerUtils.LOG.isLoggable(Level.FINE) ? new Histogram() : null;
+    var histogram = PerUtils.LOG.isDebugEnabled() ? new Histogram() : null;
```

Comment:
Whether or not we will log is decided up-front before creating `new Histogram()`. Once the instance is created, it is clear we will log whatever it will have collected at the end.